openedx/core/lib/block_structure/manager.py

```diff
@@ -30,23 +31,39 @@ def __init__(self, root_block_usage_key, modulestore, cache):
         self.modulestore = modulestore
         self.block_structure_cache = BlockStructureCache(cache)
 
-    def get_transformed(self, transformers):
+    def get_transformed(self, transformers, starting_block_usage_key=None):
         """
         Returns the transformed Block Structure for the root_block_usage_key,
-        getting block data from the cache and modulestore, as needed.
+        starting at starting_block_usage_key, getting block data from the cache
+        and modulestore, as needed.
 
-        Details: Same as the get_collected method, except the transformers'
+        Details: Similar to the get_collected method, except the transformers'
         transform methods are also called.
 
         Arguments:
             transformers (BlockStructureTransformers) - Collection of
                 transformers to apply.
 
+            starting_block_usage_key (UsageKey) - Specifies the starting block
+                in the block structure that is to be transformed.
+                If None, root_block_usage_key is used.
+
         Returns:
             BlockStructureBlockData - A transformed block structure,
-                starting at self.root_block_usage_key.
+                starting at starting_block_usage_key.
         """
         block_structure = self.get_collected()
+        if starting_block_usage_key:
+            # Override the root_block_usage_key so traversals start at the
+            # requested location.  The rest of the structure will be pruned
+            # as part of the transformation.
+            if not block_structure.has_block(starting_block_usage_key):
```

Comment:
Nit: This would be another good place for making block structures behave like fluent python objects.  Defining a __contains__ method (synonymous with has_block), would allow you to write `if usage_key not in block_structure`.